src/solvers/mod.rs

```diff
@@ -123,3 +126,12 @@ impl<N: Into<f64> + Clone> Solution for HashMap<Variable, N> {
         self[&variable].clone().into()
     }
 }
+
+/// The dual value measures the increase in the objective function's value per unit
+/// increase in the variable's value. The dual value for a constraint is nonzero only when
+/// the constraint is equal to its bound. Also known as the shadow price
+///
+pub trait SolutionWithDual: Solution {
+    /// Get a dual value of a problem for a given ConstraintReference
+    fn get_dual_value(&self, c: ConstraintReference) -> f64;
```

Comment:
If the underlying solver only offers a function to compute all dual variables together, this forces to allocate an array for all the dual results, and then throw away everything but the variable you are interested in.

What I was suggesting was something that would let you write

```
let constraint_dual : f64 =  solution.get_dual().value(my_constraint)
```

That is, first compute all the dual values, then extract the one you are interested in.

```rust
```